### ISSUE TITLE
Direction movement speed becomes butchered after trying to push a wall/object/colliding into a wall

### ISSUE THREAD

**Issue by TreyLina** [whitelilydragon/ShangMuArchitect#113]
Push a wall/object until you get the pushing sprite for a while. Then move the other direction right after. You'll notice that it takes far longer to gain foot speed.

This can also be done in midair after hitting a wall for a while. Super easy to replicate!

**Comment by TreyLina** [comment]
Ok, this bug is worse than I thought.

It doesn't just happen with changing directions. Try walking into a ceiling pillar with an open bottom, and keep moving towards the wall and hold that direction.

OR go upwards, hit a wall, and keep pressing that direction towards it. Your speed will also be butchered for a bit. It also works if you were pushing the wall and jump into midair.

TL;DR: You don't even need to switch directions to trigger this issue.

**Comment by whitelilydragon** [comment]
I was able to fix this by having the acceleration boost timer reset upon contact with walls of all surface types!

https://user-images.githubusercontent.com/23007238/107978889-8e17e500-6f8b-11eb-93fd-3c4e76628a8a.mov

Thank you for reporting this issue, this is easily one of the most important observations about FP1's behavior that I *never* noticed! I'm glad that you caught it.
